### ISSUE TITLE
Memory location inconsistent between platforms

### ISSUE THREAD

**Issue by Haptein** [citra-emu/citra#4767]
Trying to apply a couple cheats for Monster Hunter XX I noticed that they work just fine on Windows but don't in Linux(Manjaro). This doesn't seem to affect all games tho. I'm using recent nightlies in both platforms.

The memory locations of the cheats mentioned are the following:

|  Cheat    | 3ds/Windows | Linux  | Diff  |
|-----------|-------------|--------|-------|
|  60 FPS    | 88E584      | 884C5C | -9928 |
| Transmog  | AE2F0C      | AD88E4 | -A628 |

Thought it could be a dupe of  [#2127](https://github.com/citra-emu/citra/issues/2127) but was advised to open a new issue since it appears to be outdated.

**Comment by wwylele** [comment]
First of all I want to make sure that this isn't infected by some obvious factors, such as different game version (do you have CIA installed), different region setting, different game save or simply the game doesn't have deterministic memory layout itself. After ruling out these possibilities, one can start doing research by logging all svcControlMemory calls and compare the data on the two platform. The log should be inserted in the function https://github.com/citra-emu/citra/blob/46b015befd45bc74dd5ac316ed508bb832fb5d9d/src/core/hle/kernel/svc.cpp#L189 . There is already a debug level log that you can turn on, but what's more important, is to add the log of the function output on `out_addr` as well.

**Comment by wwylele** [comment]
Are the hex numbers you listed in the table memory addresses? Because if they are, these are in the *static* memory region (.text/.rodata/.data/.bss), whose layout is the "most fixed" one among all memory regions and is allocated only on booting games. Without considering internal dynamic layout (which isn't likely for the static region anyway), the only thing that can affect the layout of this region is the game binary, including game version change.

**Comment by Haptein** [comment]
You are completely right, it appears these codes were made with the latest game update in mind (which I never installed in my Linux pc). I'm sorry >.<
